tiny_dnn/io/caffe/layer_factory_impl.h

```diff
@@ -244,15 +244,15 @@ inline std::shared_ptr<layer> create_pooling(const caffe::LayerParameter &layer,
   }
 
   if (w_pad != 0) {
-    if (w_pad == pool_size_w - 1) {
+    if (w_pad == pool_size_w - 1 || w_pad == (pool_size_w - 1) / 2) {
```

Comment:
Sure, I can change it, if everyone agrees. Or you can.

Should `w_pad == pool_size_w - 1` lead to `padding::full` then?